### ISSUE TITLE
Create new circulating supply API endpoint and modify our subdomain

### ISSUE THREAD

**Issue by divine-comedian** [Giveth/giv-supply-function#5]
Currently our subdomain endpoint at https://circulating.giveth.io/token-supply is providing the latest ciculating supply info for coingecko and CMC - I believe this is being handled by an AWS service created by @marthendalnunes 

We should move this process to be handled by our own internal devOps services... I'm not sure what needs to happen to achieve this though, it would require investigation... @geleeroyale 

when we have a new endpoint that provides the required data we should reconfigure the circulating.giveth subdomain to this new internally managed endpoint and avoid and disruption of the data we provided to these external services.

cc: @laurenluz

**Comment by geleeroyale** [comment]
Should not be too crazy. I just know we use some AWS edge functions. Will ask @marthendalnunes how he implemented it

**Comment by laurenluz** [comment]
Hey @geleeroyale any update here?

**Comment by geleeroyale** [comment]
Yes - I can not reach @marthendalnunes for support and I think this was not created on our aws resources? I can not find it anywhere. Showing the total and circulating supply is working currently, but I agree that we should be able to service it.

Btw - I noticed on coinmarketcap that there are warnings about supply data for giv token. We might want to talk to them again.

**Comment by laurenluz** [comment]
I asked Griff to try to contact Vitor.

**Comment by laurenluz** [comment]
@geleeroyale 

![image](https://github.com/user-attachments/assets/966d6fd3-bcf0-4960-bef9-1ad588389fff)

is this something you can manage?

**Comment by geleeroyale** [comment]
I never did this - so its hard to judge complexity. Ill give it a go

**Comment by laurenluz** [comment]
keep us posted here! Thanks @geleeroyale

**Comment by geleeroyale** [comment]
Had no time for this yet - as I could not find good resources/examples online. I still think it would be best if we just port @marthendalnunes code to our aws resources

**Comment by laurenluz** [comment]
@marthendalnunes is super unresponsive. Several of us have tried to reach him and no answer. I asked Griff to reach out to him several times too and he just says that we should do it ourselves.

@geleeroyale is it better to delegate to someone else from devops?

**Comment by geleeroyale** [comment]
> @marthendalnunes is super unresponsive. Several of us have tried to reach him and no answer. I asked Griff to reach out to him several times too and he just says that we should do it ourselves.
> 
> @geleeroyale is it better to delegate to someone else from devops?

I am not sure why devops is being tagged to be honest.

If we have the software developed, ready to deploy -> devops
If we need to develop the software -> development

Edit: Please keep in mind that devops consists of three people. We try to avoid feature development that lies outside of operational stuff. I am not sure about the count of devs we currently have, but it must be multiple times the devops team size.

**Comment by wmb81321** [comment]
Regarding on CMC Circulating Supply could be fixed when we get access to the [CMC's self-reporting dashboard](https://docs.google.com/presentation/d/1p0-EjHE4ELpLZ8kwzP855L5x3uq941oUpCfIYSraa0g/edit#slide=id.g88e0475fa0_0_3). 

We are in the social verification step, then we will get access and fix something on Giveth's CMC page, even Circulating Supply is possible and faster

**Comment by geleeroyale** [comment]
> Regarding on CMC Circulating Supply could be fixed when we get access to the [CMC's self-reporting dashboard](https://docs.google.com/presentation/d/1p0-EjHE4ELpLZ8kwzP855L5x3uq941oUpCfIYSraa0g/edit#slide=id.g88e0475fa0_0_3).
> 
> We are in the social verification step, then we will get access and fix something on Giveth's CMC page, even Circulating Supply is possible and faster

thanks for joining the conversation!

we would like to redeploy on our own aws resources to be able to maintain it better in the future - it would also relieve you from your responsibilities in that regard.

Can you share code & deployment details with me?

**Comment by laurenluz** [comment]
@geleeroyale just checkin in here! last week you mentioned that Vitor replied so that we could port his code in to our AWS resources - got an update?

**Comment by wmb81321** [comment]
We are working to get access to the **Self-Reporting Dashboard** of Coinmarketcap to 4 accounts. We sent last week the social verification (https://x.com/Giveth/status/1818662824647835768) via form and via mail, we are waiting response.

**Comment by geleeroyale** [comment]
> [@geleeroyale](https://github.com/geleeroyale) just checkin in here! last week you mentioned that Vitor replied so that we could port his code in to our AWS resources - got an update?

I was mistaken - it was a comment by @wmb81321 that I saw

**Comment by laurenluz** [comment]
@mohammadranjbarz can you guide @Reshzera on this? or can you take it?

**Comment by wmb81321** [comment]
> We are working to get access to the **Self-Reporting Dashboard** of Coinmarketcap to 4 accounts. We sent last week the social verification (https://x.com/Giveth/status/1818662824647835768) via form and via mail, we are waiting response.
> 
> I will be here looking forward when the URL API for circulating supply and total supply when gets updated.

Furthermore, we already get access to Self-Repoorting Dashboard of CoinMarketCap! 🚀
Meanwhile, I will prepare the form to update Circulating supply on CoinMarketCap and i will add the new API URL 
 Circulating Supply when is ready 🫡

**Comment by mhmdksh** [comment]
Hey guys, @laurenluz @divine-comedian I am joining this conversation, in an attempt to release this into a new deployable way and make this easier for us to manage. 

I have already been able to run the code locally as you can see here:
Coingecko:
<img width="1283" alt="image" src="https://github.com/user-attachments/assets/879bb810-2893-47a3-b50d-1119317d3349">

CoinmarketCap:
<img width="842" alt="image" src="https://github.com/user-attachments/assets/2a62bf5f-6d48-4ae5-8acf-43578b284eff">

Which already gave the results that we needed (Please ignore the current errors). And I am attempting to release this deployment from specifically being able to deploy it on lambda. I will update you guys on this.

**Comment by mhmdksh** [comment]
Ok guys, @divine-comedian @laurenluz @geleeroyale, I got really excited to make that happen during the weekend. And I came up with a MAJOR PR to achieve this. https://github.com/Giveth/giv-supply-function/pull/7

This PR did a MAJOR refactor to how the code runs, it did the below:
1. Removed the dependency to deploy this on AWS Lambda and make it a necessary provider, which in turn will save us a lot of money hosting this endpoint
2. Made the application run as a standalone instance while including both Coinmarketcap and Coingecko criteria accessible from different endpoint paths: `giv-supply-cg` & `giv-supply-cmc`
3. Created a Dockerized version of this endpoint that can be hosted anywhere
4. Updated the packages needed to run this app (reduced the package dependency)

You can see a running example for this new release on the URLs below:

Coingecko Endpoint: http://80.240.30.129:3000/giv-supply-cg
Coinmarketcap Circulating Supply: http://80.240.30.129:3000/giv-supply-cmc?q=circulating_supply
Coinmarketcap Total Supply: http://80.240.30.129:3000/giv-supply-cmc?q=total_supply

I'll need a quick verification from someone from backend, @mohammadranjbarz Maybe?

Once this is approved, we can go on with deploying it under our domain and update the CMC and CG verification for GIV
CC: @jainkrati

**Comment by geleeroyale** [comment]
@mhmdksh Yezzzzz! Ultra-awesome man 😎

**Comment by mohammadranjbarz** [comment]
I just came here after Endaoment release to see what should we don, I saw @mhmdksh have done that in the best way, thanks man, you are a hero!!!

**Comment by mhmdksh** [comment]
Thanks for taking a look @mohammadranjbarz , I merged the changes.
Our next step is deployment, I will update you guys once this is ready to send out to CMC & CG

**Comment by geleeroyale** [comment]
are you ready to get the domain pointed your way @mhmdksh ?

**Comment by mhmdksh** [comment]
Yes @geleeroyale can you configure this name record:

`supply.giveth.io` --> `46.101.125.170`

Or choose any similar domain.

**Comment by geleeroyale** [comment]
Record is set

**Comment by mhmdksh** [comment]
Thanks @geleeroyale

# Update
@divine-comedian @laurenluz @wmb81321 

We now have new endpoints for the circulating supply working on the new app that we modified.
It is now hosted on our infrastructure under the below links:

### Coingecko:
Circulating + Total Supplies: https://supply.giveth.io/giv-supply-cg

### Coinmarketcap:
Circulating Supply: https://supply.giveth.io/giv-supply-cmc?q=circulating_supply
Total Supply: https://supply.giveth.io/giv-supply-cmc?q=total_supply

We can close this issue if nothing else is required

**Comment by laurenluz** [comment]
Hey @mhmdksh is there anyway we can make the URLs the same? or have the old one reroute to the new one?

otherwise it will a bunch of admin work for us to try to update them on coingecko

**Comment by mhmdksh** [comment]
@laurenluz Yes of course, this can be achieved, we need a little tinkering though, I'll push another PR for that change to happen

**Comment by laurenluz** [comment]
yeah!! ty!

**Comment by mhmdksh** [comment]
@laurenluz @divine-comedian I have integrated the updated version of this endpoint to the same old URLs, and they are now reflecting the correct information under those same old links:

### Coingecko:
Circulating + Total Supplies: https://circulating.giveth.io/token-supply

### Coinmarketcap:
Circulating Supply: https://supply.giveth.io/giv-supply-cmc?q=circulating
Total Supply: https://supply.giveth.io/giv-supply-cmc?q=totalcoins

Please check the status now, and let me know if these are reflected with Coingecko and Coinmarketcap

**Comment by divine-comedian** [comment]
@laurenluz @wmb81321 can we check if this issue is resolved and if so, close it?

**Comment by laurenluz** [comment]
yeah it's not updated yet, on my list... or @wmb81321 if you can try to update it on CMC and check, please do.

**Comment by wmb81321** [comment]
I sent a request with the `Self-reporting Dashboard `of CMC, maybe will take some 1-3 days to get and answer.

<img width="834" alt="Screenshot 2024-09-03 at 1 09 56 PM" src="https://github.com/user-attachments/assets/bd94170b-ad0c-4ce4-a672-e77bc27ba147">

<img width="832" alt="Screenshot 2024-09-03 at 1 18 06 PM" src="https://github.com/user-attachments/assets/48887d5b-41e6-43de-8e00-67c47a2fe492">

**Comment by laurenluz** [comment]
thank you @wmb81321! please keep an eye on it and let me know once it's accepted & live!

I think we can close this issue now, since the API endpoint & subdomain is done, but I'll make another issue to track the update to CMC

**Comment by wmb81321** [comment]
Got it, we will update the updating circulating and total supply on CMC here: https://github.com/Giveth/giveth-dapps-v2/issues/4677#event-14122169225
